tests/test_single_topology.py

```diff
@@ -450,7 +455,8 @@ def get_max_distance(x0):
             # assert that the idxs are canonicalized.
             assert bond_idxs_are_canonical(system.bond.potential.idxs)
             assert bond_idxs_are_canonical(system.angle.potential.idxs)
-            assert bond_idxs_are_canonical(system.torsion.potential.idxs)
+            assert bond_idxs_are_canonical(system.proper.potential.idxs)
+            assert_improper_idxs_are_canonical(system.improper.potential.idxs)
```

Comment:
nit[reduced mental overhead + name-mangling]: could have each of these lines be `assert <potential_type>_idxs_are_canonical(system.<potential_type>.potential.idxs)` e.g. by defining aliases `nonbonded_idxs_are_canonical proper_idxs_are_canonical = angle_idxs_are_canonical = chiral_bond_idxs_are_canonical = bond_idxs_are_canonical ...` and tweaking `assert_improper_idxs_are_canonical` to `improper_idxs_are_canonical` . Then, a reader might safely skim over these repetitive lines, and just check whether the definitions are reasonable. 

Tangent: The general logic of the new function `assert_improper_idxs_are_canonical` (assert `idxs == canonicalize(idxs)` seems clearer than the general logic of `bond_idxs_are_canonical` (check first_idx < last_idx).

Reply:
I need to do this more systematically than just this file re: canonicalizing bonds - there's at least 3 different repetitive definitions of `canonicalize_bond` in `dummy.py, single_topology.py, and ff/handlers/utils.py` - okay if I defer this to a separate deboggling task? Really trying to minimize clerical changes for this PR to avoid a blow-up of diffs like #1420